src/components/MovieCard.js

```diff
@@ -1,19 +1,71 @@
 import "./MovieCard.css";
-import React from "react";
+import React, {useEffect, useRef, useState, Suspense} from "react";
 import PropTypes from "prop-types";
 
-const  MovieCard = ({movie}) => (
-    <div className="movie-card">
-        <img src={require(`../static/images/movies/${movie.icon}`)}/>
-        <div className="movie-card-info">
-            <div className="movie-card-name">{movie.name}</div>
-            <div className="movie-card-genres">
-                {movie.genres.map(genre=>genre.name).join(", ")}
+const ContextMenu = React.lazy(()=> import("./ContextMenu"));
+
+
+const  MovieCard = function(props){
+    const [anchorPoint, setAnchorPoint] = useState({ x: 0, y: 0 });
+    const [showContextMenu, setShowContextMenu] = useState(false);
+
+    const closeContextMenu = ()=>{
+        setShowContextMenu(false)
+    }
+
+    const movie = props.movie;
+    if(movie === undefined){
+        movie = {};
+    }
+    if(movie.name === undefined){
+        movie.name = "Best Movie";
+    }
+    if(movie.year === undefined){
+        movie.year = 2000;
+    }
+    if(movie.icon === undefined){
+        movie.icon = "default.png";
+    }
+    if(movie.genres === undefined){
+        movie.genres = {}
+    }
+
+    const innerRef = useRef(null);
+    useEffect(() => {
+        function handleContextMenu(event) {
+            event.preventDefault();
+            setAnchorPoint({ x: event.pageX, y: event.pageY });
+            setShowContextMenu(true);
+        }
+        const div = innerRef.current;
+        div.addEventListener("contextmenu", handleContextMenu);    
+        return () => {
+          div.removeEventListener('contextmenu', handleContextMenu);
+        };
+      }, []);
+
+    return (
+        <>
+            <div className="movie-card" ref={innerRef}>
+                <img src={require(`../static/images/movies/${movie.icon}`)}/>
+                <div className="movie-card-info">
+                    <div className="movie-card-name">{movie.name}</div>
+                    <div className="movie-card-genres">
+                        {movie.genres.map(genre=>genre.name).join(", ")}
+                    </div>
+                    <div className="movie-card-year">{movie.year}</div>
+                </div>
             </div>
-            <div className="movie-card-year">{movie.year}</div>
-        </div>
-    </div>
-)
+            <Suspense fallback={<div className="error">Loading...</div>}>
```

Comment:
Suspense is experimental component and I think that is not good idea to use it in educational project

Reply:
it was in the lecture, that why it is used - so I have it on record not to forget it if I need it in a future